cmd/srcd-server/engine/components.go

```diff
@@ -0,0 +1,29 @@
+package engine
+
+import "github.com/src-d/engine-cli/docker"
+
+// Component to be run.
+type Component struct {
+	Name         string
+	Start        docker.StartFunc
+	Options      []docker.ConfigOption
+	Dependencies []Component
+}
+
+// Run the given components if they're not already running. It will recursively
+// run all the component dependencies.
+func Run(cs ...Component) error {
+	for _, c := range cs {
+		if len(c.Dependencies) > 0 {
+			if err := Run(c.Dependencies...); err != nil {
```

Comment:
I'm afraid this might lead to infinite loops.

If `A` depends on `B` and `B` on `A` will this recursive call ever finish?